README_extending_docker.md

```diff
@@ -63,33 +63,19 @@ You can use our base docker to extend the functionality to fit your need. The st
 2. Push your `custom docker` to any docker registry for redistribution.
 <br/>
 
+## Actions and agruments
 
+Actions are small python functions that is designed to do a specific task. For example, aws_sts_get_caller_identity action
+is designed to display the  AWS sts caller identity for a given configuration. Actions may take one or more arguments, like 
+any python function do. Some or all of these arguments may also assume a default value if none given at the time of calling.
+Many checks may have the same argument name used. For example `region` could be a common name of the argument used across
+multiple AWS actions, likewise `namespace` could be a common argument for an K8S action. 
 
-## How to Copy Custom Actions and Runbook
 
-If you have deployed our Awesome runbook as a Kubernetes POD then follow the step below
-1. Copy the custom actions from the POD to your local machine so you can bundle into your custom Docker for re-distribution
-```
-kubectl cp <AWESOME_POD_NAME>:/unskript/data/runbooks -n <NAMESPACE> $YOUR_REPO_DIRECTORY/runbooks
-kubectl cp <AWESOME_POD_NAME>:/unskript/data/actions -n <NAMESPACE> $YOUR_REPO_DIRECTORY/actions
-cd $YOUR_REPO_DIRECTORY
-git clone https://github.com/unskript/Awesome-CloudOps-Automation.git
-
-Example:
-
-kubectl cp awesome-runbooks-0:/unskript/data/actions -n awesome-ops $YOUR_REPO_DIRECTORY/actions
-kubectl cp awesome-runbooks-0:/unskript/data/runbooks -n awesome-ops $YOUR_REPO_DIRECTORY/runbooks
-```
-
-If you have deployed our Awesome runbook as a Docker instance, then you can use
-the following step.
-```
-export CONTAINER_ID=`docker ps | grep awesome-runbooks | awk '{print $1}'`
-docker cp $CONTAINER_ID:/unskript/data/actions $HOME/Workspace/acme/actions
-docker cp $CONTAINER_ID:/unskript/data/runbooks $HOME/Workspace/acme/runbooks
-```
-
-Lets consider two scenarios as starting point for extending Awesome docker
+We call an action a check (short for health check) when the return value of the action is in the form of a Tuple.
+First value being the result of the action (a boolean) and the second value being the errored object or None. 
```

Comment:
the result of the check , whether it passed or not. True being check passed, False otherwise.